app/src/main/java/org/akvo/flow/util/SurveyIdGenerator.java

```diff
@@ -33,12 +33,15 @@ public SurveyIdGenerator() {
      * Get the survey id from the survey xml folder
      * The structure can be either surveyId/ or surveyId/folder/
      *
-     * @param folderPath all the folder path expluding the actial fileName
+     * @param entryName all the folder path expluding the actial fileName
      * @return
      */
     @NonNull
-    public String getSurveyIdFromFilePath(@Nullable String folderPath) {
-        String folders[] = folderPath == null ? null : folderPath.split("/");
+    public String getSurveyIdFromFilePath(@Nullable String entryName) {
+        int fileSeparatorPosition = entryName == null? -1 : entryName.lastIndexOf("/");
```

Comment:
Ditto here re `indexOf()`

And if the setup is always `surveyId/folder`, then I would pick everything up to the first "/" and take that as the id. And if we can't figure out the id, return a `null`